packaging/setup/plugins/ovirt-engine-common/ovirt-engine-keycloak/core/admin_user.py

```diff
@@ -0,0 +1,85 @@
+
+# ovirt-engine-setup -- ovirt engine setup
+#
+# Copyright oVirt Authors
+# SPDX-License-Identifier: Apache-2.0
+#
+#
+
+"""Ovirt engine keycloak initial admin user plugin."""
+
+import datetime
+import gettext
+import os
+
+from otopi import plugin
+from otopi import util
+
+from ovirt_engine_setup import constants as osetupcons
+from ovirt_engine_setup import util as osetuputil
+from ovirt_engine_setup.engine import constants as oenginecons
+from ovirt_engine_setup.engine_common import constants as oengcommcons
+from ovirt_engine_setup.keycloak import constants as okkcons
+
+
+def _(m):
+    return gettext.dgettext(message=m, domain='ovirt-engine-keycloak')
+
+
+@util.export
+class Plugin(plugin.PluginBase):
+    """Ovirt engine keycloak initial admin user plugin."""
+
+    def __init__(self, context):
+        super(Plugin, self).__init__(context=context)
+
+    @plugin.event(
+        stage=plugin.Stages.STAGE_MISC,
+        condition=lambda self: (
+                self.environment[
+                    oenginecons.CoreEnv.ENABLE
+                ] and
+                self.environment[
+                    oenginecons.EngineDBEnv.NEW_DATABASE
+                ]
+
+        ),
+    )
+    def _create_admin(self):
+        password = self.environment.get(oenginecons.ConfigEnv.ADMIN_PASSWORD)
+        if password:
+            self.logger.info(_('Creating initial Keycloak admin user'))
+            self._backup_existing_add_user_json_file()
+            self.execute(
+                (
+                    okkcons.FileLocations.KEYCLOAK_ADD_USER_SCRIPT,
+                    '-r', okkcons.Const.KEYCLOAK_MASTER_REALM,
+                    '-u', self.environment[
+                        oenginecons.ConfigEnv.ADMIN_USER
+                    ].rsplit('@', 1)[0],
+                    '-p', password,
+                    '--sc', okkcons.FileLocations.OVIRT_ENGINE_CONFIG_DIR,
+                ),
+            )
+            os.chown(
+                okkcons.FileLocations.KEYCLOAK_ADD_INITIAL_ADMIN_FILE,
```

Comment:
It's unfortunate, but the file is created by the keycloak add user script and the target location is where wildfly server config directory lives  (--sc  option).

Reply:
Actually I was wrong, --sc option ('server config') is misleading because I can point it to anywhere and the file is created there.  It is possible to do as you suggest - create secure tmp directory, generate file there, setup it's permissions and move it to target location.

Reply:
I will handle it in the following patches